lte/gateway/python/magma/pipelined/app/enforcement_stats.py

```diff
@@ -347,20 +363,17 @@ def _handle_flow_stats(self, stats_msgs):
 
         self.logger.debug("Processing %s stats responses", len(stats_msgs))
         # Aggregate flows into rule records
-        current_usage = defaultdict(RuleRecord)
+        aggregated_msgs = []
         for flow_stats in stats_msgs:
-            self.logger.debug("Processing stats of %d flows", len(flow_stats))
-            for stat in flow_stats:
-                if stat.table_id != self.tbl_num:
-                    # this update is not intended for policy
-                    return
-                try:
-                    current_usage = self._update_usage_from_flow_stat(
-                        current_usage, stat)
-                except ConnectionError:
-                    self.logger.error('Failed processing stats, redis unavailable')
-                    self.unhandled_stats_msgs.append(stats_msgs)
-                    return
+            aggregated_msgs += flow_stats
+
+        self.logger.debug("Processing stats of %d flows", len(aggregated_msgs))
+        try:
+            current_usage = self._get_usage_from_flow_stat(aggregated_msgs)
+        except ConnectionError:
+            self.logger.error('Failed processing stats, redis unavailable')
+            self.unhandled_stats_msgs.append(stats_msgs)
+            return
```

Comment:
can you explain why this refactoring change is done in this commit ?

Reply:
I've seen the yield behave slightly worse that just looping through the function